.github/actions/pmd/action.yml

```diff
@@ -0,0 +1,46 @@
+name: "PMD"
+description: "Run PMD against changes in a pull request"
+inputs:
+  pmd-version:
+    description: The version of PMD to use
+    required: false
+    default: "6.55.0"
+  create-github-annotations:
+    description: |
+      Whether to create inline comments on the PR using GH Advanced Security. This is free for open source projects but
+      requires a license for private repos.
+    required: false
+    default: "true"
+  fail-on-new-issues:
+    description: |
+      When set to true this fails the build if the PR introduces more issues than it resolves.
+    required: false
+    default: "true"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+      env:
+        REF_TO_CHECKOUT: ${{ inputs.auto-commit == 'true' && github.head_ref || '' }}
+      with:
+        ref: ${{ env.REF_TO_CHECKOUT }}
```

Comment:
I'm not quite sure what's this section here for? I don't see any `auto-commit` inputs in this action.

Reply:
not sure either what env.REF_TO_CHECKOUT is for, usually it should be smart enough without parameters

Reply:
It was probably in whatever action I copied to get started! I'll simplify this to the defaults.

Reply:
Usually we don't even include the checkout step in our actions at all, it's assumed that the user checks out the ref they're interested in. I think you might have copied this from an action which has an edge-case such as it wants to commit things on a specific branch in case of auto-commit being true (e.g.: the pre-commit action).

Personally I'd get rid of the whole step.